services/py-api/src/service/participants_service.py

```diff
@@ -35,10 +38,29 @@ async def _create_participant_and_team_in_transaction(
             return participant
 
         return Ok((participant.ok_value, team.ok_value))
+    
+    async def _check_capacity(self) -> bool:
+        """Calculate if there is enough capacity to register a new team"""
+        # Fetch number of verified random participants
+        verified_random_participants = await self._participant_repo.get_verified_random_participants_count()
+
+        # Fetch number of verified registered teams
+        verified_registered_teams = await self._team_repo.get_verified_registered_teams_count()
+
+        # Calculate the anticipated number of teams
+        number_ant_teams = verified_registered_teams + math.ceil(verified_random_participants / self.team_capacity)
+
+        # Check against the hackathon capacity
+        return number_ant_teams < self.hackathon_cap
 
     async def register_admin_participant(
         self, input_data: ParticipantRequestBody
-    ) -> Ok[Tuple[Participant, Team]] | Err[DuplicateEmailError | DuplicateTeamNameError | Exception]:
-        # TODO: Add Capacity check 2
+    ) -> Ok[Tuple[Participant, Team]] | Err[DuplicateEmailError | DuplicateTeamNameError| CapacityExceededError | Exception]:
+        # Capacity Check 2
+        has_capacity = await self._check_capacity()
+        if not has_capacity:
+            return Err(CapacityExceededError("Hackathon capacity exceeded, cannot register new team"))
```

Comment:
return just the max capacity in the body

Reply:
Fixed f33e8f7a85dbefbcda9a089a8ba5a1d3e7e93e9b